### DISCUSSION TITLE
git bash and oauth tokens

### DISCUSSION THREAD

**Discussion by ManySolutions** [community/community#38064] [New to GitHub]
I have googled this and gone through pretty much ALL of the results on the first two pages with absolutely no success.

I am trying to run a laravel project on windows server. I am trying to run php composer install, i get the error:
`[UnexpectedValueException]
  Your github oauth token for github.com contains invalid characters:`

I have tried different tokens from github and lo luck. I have tried the beta tokens, no luck.
I have tried to add my credentials via global: 'git config --global user.name "your name" '
No luck.
I cloned the repo using github desktop in the hopes it would somehow allow me to run composer without issues.

No luck.
There were even results saying update composer to 2.2.6, but the website says 2.2.4 is the latest version. 

I have never been so frustrated in my life.

**Comment by david-engelmann** [comment]
Are you passing the oauth token as an env variable. I've noticed special characters having issue when being passed in the past. Also, make sure you are using your app password

**Reply by ManySolutions** [reply]
I used:
$ composer config -g github-oauth.github.com gitkey  to pass the key.  The key was from github.com

**Comment by samus-aran** [comment]
Welcome to the GitHub Community, @ManySolutions, we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!

**Comment by airtower-luna** [comment]
> I have tried to add my credentials via global: 'git config --global user.name "your name" '

I can't comment on Composer, but I want to mention that putting credentials into `user.name` or `user.email` is not only not going help, but dangerous: You'd be publishing your credentials with each commit. Make sure to set it back to the name you want recorded as author in your commits!
